internal/rbd/snapshot.go

```diff
@@ -400,3 +400,48 @@ func (rbdSnap *rbdSnapshot) SetVolumeGroup(ctx context.Context, cr *util.Credent
 
 	return nil
 }
+
+// GetVolSize returns the size of the snapshot in bytes.
+func (rbdSnap *rbdSnapshot) GetVolSize() int64 {
+	return rbdSnap.VolSize
+}
+
+// rbdSnapFromSnapshot converts a types.Snapshot to a *rbdSnapshot.
+// It returns an error if the snapshot is nil or not of type *rbdSnapshot.
+func rbdSnapFromSnapshot(snap types.Snapshot) (*rbdSnapshot, error) {
```

Comment:
can we add a function to Manager interface to return `rbdSnapshot` we dont need to have implementation here we can directly pass `*rbdSnapshot` to the `StreamMetadata` method

Reply:
> can we add a function to Manager interface to return `rbdSnapshot` we dont need to have implementation here we can directly pass `*rbdSnapshot` to the `StreamMetadata` method


>*rbdSnapshot to the StreamMetadata method
this was the implementation in one of my earlier iteration

The current approach was suggested by @nixpanic in 

https://github.com/ceph/ceph-csi/pull/5347#discussion_r2128818985